### PR TITLE
[CINN]Add OriginalAttributesFilter for grad op shape cache

### PR DESCRIPTION
**Pull request by Hongqing-work**
### PR Category

CINN

### PR Types

Improvements

### Description

Pcard-67164
This PR addsAdd OriginalAttributesFilter  which solves the backward op infer shape cache miss problem.

### CODE REVIEW

File: paddle/fluid/pir/dialect/op_generator/cache_grad_op_symbol_shape_gen.py [hunk 1]
```diff
@@ -37,11 +37,23 @@
 }}  // namespace paddle
 """
 
+GET_VALID_ATTR_MAP_FUNC_CODE_TEMPLATE = """
+std::unordered_map<std::string, std::unordered_set<std::string>> GetValidAttributeMapForShapeCache() {{
```
Comment (zyfncg) [inline paddle/fluid/pir/dialect/op_generator/cache_grad_op_symbol_shape_gen.py]: 返回结果为什么不用`std::unordered_map<std::string, std::set<std::string>>`

File: paddle/pir/src/dialect/shape/utils/shape_analysis.cc [hunk 2]
```diff
@@ -900,4 +872,43 @@ void InferSymbolicShapeCacheKey::SetInputShapeOrDatas(
...
+
+  const auto& IsValidAttrName = [&](const std::string& op_name,
+                                    const std::string& attr_name) -> bool {
+    static const char* kOpCallStack = "op_callstack";
+    static const char* kSymShapeStr = "sym_shape_str";
+    static const char* kResultName = "name";
+    static const char* kStopGradient = "stop_gradient";
+    if (attr_name == kOpCallStack || attr_name == kSymShapeStr ||
+        attr_name == kStopGradient || attr_name == kResultName)
+      return false;
```
Comment (zyfncg) [inline paddle/pir/src/dialect/shape/utils/shape_analysis.cc]: 有了attribute名单，这些特殊情况是不就可以不用考虑了？

File: paddle/cinn/hlir/dialect/operator/transforms/add_cinn_pass.cc [hunk 3]
```diff
@@ -99,6 +101,8 @@ void ApplyShapeOptimizationPass(
     const std::function<std::shared_ptr<::pir::PassManager>()>&
         CreatePassManager) {
   std::shared_ptr<pir::PassManager> pass_manager = CreatePassManager();
+  pir::ValidAttrFilterForShapeCache::Instance().SetValidAttributeMap(
+      paddle::dialect::GetValidAttributeMapForShapeCache());
```
Comment (Hongqing-work) [inline paddle/cinn/hlir/dialect/operator/transforms/add_cinn_pass.cc]: paddle::dialect::GetAllOpOriginalAttributes

File: paddle/fluid/pir/dialect/op_generator/cache_grad_op_symbol_shape_gen.py (hunk 1 above)
Comment (Hongqing-work) [reply paddle/fluid/pir/dialect/op_generator/cache_grad_op_symbol_shape_gen.py]: 已修改

File: paddle/pir/src/dialect/shape/utils/shape_analysis.cc (hunk 2 above)
Comment (Hongqing-work) [reply paddle/pir/src/dialect/shape/utils/shape_analysis.cc]: 最好还是保留一下，attribute名单可能没有涵盖manual的

**Review by zyfncg — APPROVED** [review]

**Review by zhangbo9674 — APPROVED** [review]
LGTM for PIR include

**Pull request merged**